### ISSUE TITLE
awslogs should respect the region in ~/.aws/config 

### ISSUE THREAD

**Issue by AlJohri** [jorgebastida/awslogs#226]
The ~/.aws/config file works in conjunction with the ~/.aws/credentials file for specifying non-credential options such as the default region.

For example:

```bash
$ cat ~/.aws/config
[default]
output = json
region = us-east-1

[bigdata]
output = json
region = us-east-1
```

**Comment by jorgebastida** [comment]
Yes, we should respect the configuration the cli uses (https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-configure.html)... and in the same order.

I was expecting boto3 (or botocore) to do so if we pass ``None`` as region. We should investigate if that is not the case and fix it.
